### ISSUE TITLE
sound glitches with FLAC

### ISSUE THREAD

**Issue by Nakilon** [mltframework/shotcut#1002]
I've spent two days on editing the 4 hours long screencast recording and now when I export the file there are two kinds of sound glitches everywhere. I thought they happen only during play in the editor and I didn't expect this in the exported file too.

My source file is:
```
General
Unique ID                                : 117483743134593995970346840894330515223 (0x58628BBF539805713E0BB21E55EE2F17)
Complete name                            : D:\temp\mix.mkv
Format                                   : Matroska
Format version                           : Version 4
File size                                : 13.0 GiB
Duration                                 : 4 h 14 min
Overall bit rate mode                    : Variable
Overall bit rate                         : 7 304 kb/s
Writing application                      : Lavf58.45.100
Writing library                          : Lavf58.45.100
ErrorDetectionType                       : Per level 1

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : Main@L5.1
Format settings                          : CABAC / 4 Ref Frames
Format settings, CABAC                   : Yes
Format settings, Reference frames        : 4 frames
Format settings, GOP                     : M=1, N=30
Codec ID                                 : V_MPEG4/ISO/AVC
Duration                                 : 4 h 14 min
Width                                    : 2 560 pixels
Height                                   : 1 440 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 60.000 FPS
Standard                                 : Component
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Default                                  : Yes
Forced                                   : No
Color range                              : Limited

Audio
ID                                       : 2
Format                                   : FLAC
Format/Info                              : Free Lossless Audio Codec
Codec ID                                 : A_FLAC
Duration                                 : 4 h 14 min
Bit rate mode                            : Variable
Channel(s)                               : 1 channel
Channel layout                           : C
Sampling rate                            : 48.0 kHz
Bit depth                                : 16 bits
Compression mode                         : Lossless
Delay relative to video                  : -452 ms
Writing library                          : Lavc58.91.100 flac
Default                                  : Yes
Forced                                   : No
```
sample: https://www.youtube.com/watch?v=dQePv7UuU5w

I tried different audio options but the result is the same and is probably not related to them at all because I hear the same in the editor. Here I provide its mediainfo too anyway:
```
General
Unique ID                                : 162390503991419540252131448940683028286 (0x7A2B45FFA5C14D07A8360BAC3273673E)
Complete name                            : D:\VIDEOS\stalkerproject - Copy.mp4
Format                                   : Matroska
Format version                           : Version 4
File size                                : 155 MiB
Writing application                      : Lavf58.29.100
Writing library                          : Lavf58.29.100
IsTruncated                              : Yes
ErrorDetectionType                       : Per level 1
FileExtension_Invalid                    : mkv mk3d mka mks

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : Main@L5
Format settings                          : CABAC / 4 Ref Frames
Format settings, CABAC                   : Yes
Format settings, Reference frames        : 4 frames
Codec ID                                 : V_MPEG4/ISO/AVC
Width                                    : 2 560 pixels
Height                                   : 1 440 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 30.000 FPS
Standard                                 : Component
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Default                                  : Yes
Forced                                   : No
Color range                              : Limited
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709

Audio
ID                                       : 2
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : A_AAC-2
Channel(s)                               : 1 channel
Channel layout                           : C
Sampling rate                            : 48.0 kHz
Frame rate                               : 46.875 FPS (1024 SPF)
Compression mode                         : Lossy
Delay relative to video                  : -21 ms
Default                                  : Yes
Forced                                   : No
```
sample: https://www.youtube.com/watch?v=0cqvcgrnRKg

Here is the project file: [stalker.mlt.txt](https://github.com/mltframework/shotcut/files/5801310/stalker.mlt.txt)

Windows 10. Shotcut 20.02.17

**Comment by ddennedy** [comment]
You are using a version that is almost a year old, and there were [fixes](https://shotcut.org/blog/) in this area since version 20.02. Please upgrade and try again.

**Comment by Nakilon** [comment]
Same: https://www.youtube.com/watch?v=8uP8c5kIC50

**Comment by Nakilon** [comment]
For a sample I've made a small version of the recording like this:
```
ffmpeg -t 120 -i d:\temp\mix.mkv -vf scale=256:144 d:\temp\small.mkv
```
and replaced all occurences of `mix.mkv` with `small.mkv`. And the glitch is gone.
Interesting that empty segments still have previews:
![image](https://user-images.githubusercontent.com/2870363/104431015-7c13e280-5598-11eb-937f-acf3964333d9.png)
Does Shotcut cache them somewhere?

Anyway.
Looks like somehow the current video format breaks the audio for Shotcut.

**Comment by Nakilon** [comment]
Here are 2 MB of files to reproduce it -- small.mkv and small.mlt 
[small.zip](https://github.com/mltframework/shotcut/files/5807690/small.zip)
[small.zip](https://github.com/mltframework/shotcut/files/5807692/small.zip)

**Comment by luzpaz** [comment]
Is this issue still relevant ?

**Comment by ddennedy** [comment]
Fixed in version 23.05.14
